Rewrite GitHub Data script to use octokit and true-github-contributors

KianBadie opened the pull request:
**Edited to reflect changes of commits since 10/3**

This pull request updates the GitHub Aggregation script in major ways:
- Removes use of the `request-promise` library in favor of GitHub's API client [octokit](https://octokit.github.io/rest.js/v18).
- Removes request to issue comment endpoints and functions that aggregate issue comments/contributions in favor of using the [true-github-contributors](https://www.npmjs.com/package/true-github-contributors) package that performs that data fetching and that aggregation.
- Implements the [lodash](https://lodash.com/) package in order to conveniently create copies of data in script so that I could conveniently store result data.

Important changes to `github-data.js`:
- All three contribution data sets (issue comments, commits, issue comments + commits) are still there. In each respective data set, the url endpoints to retrieve contribution data that used to be stored under the `urls` property will be removed when this script does it's first run. This is because the are no longer needed.
- `contributors` data set per project will be renamed to -> `commitContributors` for clarity as to what this data set actually is.
- More data will start showing up. This is because the way the old script was written, it was not fetching all the data. For example, it seems that our original script was not fetching all repositories (I have a hunch this is because the way we called the search endpoint, we only fetched maybe the first 30 repos that showed up), but now the script fetches more repos like adopt-civic-art.

Changes to `package.json` and `package-lock.json`:
- Uses `@octokit/rest` and `true-github-contributors` in favor of the `request` packages.
- Uses `lodash` library
- I believe change in packages caused the changes to `package-lock.json`. Looks like `package-lock.json` got updated by some general updates as well.

If you have any questions or comments, please feel free to message me. My goal of this rewrite was to write something that is convenient, readable, and that someone could modify in the future. So if it seems convoluted or unnecessarily complicated, please let me know.

KianBadie committed:
Write initial github script using new contributors mixin.

KianBadie committed:
Merge branch 'contributors-mixin' into update-gh-script

KianBadie committed:
Modify github script to run on push and modify file paths in script.

KianBadie committed:
change script back to run on original schedule.

cnk commented on github-actions/package.json:
```diff
@@ -1,20 +1,20 @@
-    "author": "",
-    "license": "ISC"
+  "name": "Website",
+  "version": "0.0.0",
```
I think it may be time to version this. We never did 1.0 and yet this is clearly 2.0. And please fill in the author field.

KianBadie replied:
Sounds good to me, and who should I put as the author? This file does have a long history.

KianBadie replied:
Added myself as author and updated version to 2.0.0

cnk reviewed (approved):
I like the way this was refactored to take much of the complexity out of this script and move it to it's own NPM package. This makes it much easier for me to follow what this script is trying to do. 

I am fine with this PR but will defer to @ruben1s for a more detailed review of the JavaScript.

KianBadie commented:
@cnk @ruben1s please don't merge this until after tomorrow. I am letting this script run in my fork at the same time with the same data and I want to see if the results are the same. I have been doing some other checks and the data does indeed look like expected, but I want to make sure.

cnk commented:
The author should now be Kian Badie. Previous versions had contributions
from other people but this version is all your work. Good job! Take some
credit for it. 😉

On Thursday, October 1, 2020, Kian Badie <notifications@github.com> wrote:

> *@KianBadie* commented on this pull request.
> ------------------------------
>
> In github-actions/package.json
> <https://github.com/hackforla/website/pull/757#discussion_r498551461>:
>
> > -    "dependencies": {
> -      "@actions/core": "^1.2.0",
> -      "@actions/github": "^2.0.0",
> -      "@actions/http-client": ">=1.0.8",
> -      "request": "2.88.0",
> -      "request-promise": "4.2.5"
> -    },
> -    "devDependencies": {},
> -    "scripts": {
> -      "test": "echo \"Error: no test specified\" && exit 1"
> -    },
> -    "keywords": [],
> -    "author": "",
> -    "license": "ISC"
> +  "name": "Website",
> +  "version": "0.0.0",
>
> Sounds good to me, and who should I put as the author? This file does have
> a long history.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/hackforla/website/pull/757#discussion_r498551461>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAABOQEIDKD2WM6CYS4UBBLSIUCODANCNFSM4R6OT7AQ>
> .
>


-- 
Cynthia Kiser
cynthiakiser@gmail.com

KianBadie committed:
Update script structure, modify package.json, and lodash package:
1. Fetch commit/comment contributions seperately, use to build total contributors, and store all 3 in the result data.
2. Add Kian Badie to author and make package version 2.0.0.
3. Add lodash package to make deep copies of contributors lists during script execution.

KianBadie commented:
Added commits that:
1. Fetch commit/comment contributions seperately, use to build total contributors, and store all 3 in the result data.
2. Add Kian Badie to author and make package version 2.0.0.
3. Add lodash package to make deep copies of contributors lists during script execution.
4. Change the log statements that will show up when the github actions run to show total issue comment contributions of each user rather than logging an object for each individual comment. 

The reason for item `1.` was that the way I was originally fetching contributors in this script revision was causing a bug where I wasn't fetching all contributors. The reason for item `3.` was that creating deep copies of the contributors list during scrip execution turned out to be helpful in the way I was storing data in the final result data array. Let me know if any changes are needed.

ruben1s reviewed (commented):
Everything looks good, but are the console.logs going to stay in your script?

KianBadie commented:
@ruben1s GitHub Actions provides a console where these console.logs will appear.

KianBadie commented:
Updated the description of this pr to reflect the most up to date changes it will bring.
